### DISCUSSION TITLE
Submissionの検証手順

### DISCUSSION THREAD

**Discussion by ktsujino** [smly/mjai.app#8] [Q&A] [answered]
submitしたzipファイルがどのように検証されるのか、可能であれば検証用のスクリプト等を公開いただくことは可能でしょうか？

submitが検証を通っていない状況なのですがデバッグが難しく、手元で再現ができればありがたく思います。

手元でgithubにあるDockerfileを使ってイメージをビルドし、コンテナ中で
$ python bot.py 0
としてbotを起動した上でドキュメントにある入出力例を試したのですが、手元では成功しているように見えており、サーバ内での動作と一致しているか確認したいと思っております。

**Comment by smly** [comment, marked as answer]
@ktsujino さん、フィードバックいただきありがとうございます！デバッグの難しい状態にお付き合いさせてしまい申し訳ないです。
直近の投稿ファイルの検証についてはサーバー側に問題があったのではないかと思われます。
これから調査に着手いたします。問題のない状態になりましたらこちらに報告いたします。今しばらくお待ち下さい。

簡易的ではありますがツモ切り bot と対戦させるテストケース以外のテスト例をアップロードしました。
（注意：事前にDockerfile のビルドが必要です。また外部コマンドを呼ぶスクリプトであるため、実行前に事前に内容をご確認ください）
https://github.com/smly/mjai.app/blob/main/verify_submission.py

このコードで直近でアップロードされた投稿ファイルを確認したところ、テストに問題なく通過することを確認しました。
お手数をおかけしてしまい申し訳ないです。

**Reply by ktsujino** [reply]
@smly さん、ご調査をいただきありがとうございます。手元で検証してみます。
対戦できるようになるのを楽しみにしております!

**Reply by smly** [reply]
@ktsujino 原因がわかりました。対戦サーバーと検証用サーバーの Docker イメージが異なっており、検証サーバーの Docker イメージには mjx がインストールされていないことが原因でした。申し訳ありませんでした。

イメージのビルドと設定を完了し次第、投稿ファイルの再評価いたします。対戦を楽しみにしております！

**Reply by ktsujino** [reply]
再検証が行われ動作しているのを確認いたしました。素早いご対応まことにありがとうございました!
